plugins/cd/__init__.py

```diff
@@ -25,40 +25,44 @@
 # from your version.
 
 import dbus
-from fcntl import ioctl
 import logging
-import os
-import struct
+import os.path
+import sys
+
+from gi.repository import GLib
 
 from xl.nls import gettext as _
-from xl import providers, event
+from xl import providers, event, main
 from xl.hal import Handler, UDisksProvider
 from xl.devices import Device, KeyedDevice
-
-from xl import playlist, trax, common
-import os.path
+from xl import playlist, trax, common, settings
+from xl.trax import Track
 
 import cdprefs
+import _cdguipanel
 
-try:
-    import DiscID
-    import CDDB
-
-    CDDB_AVAIL = True
-except Exception:
-    CDDB_AVAIL = False
 
 logger = logging.getLogger(__name__)
 
 
-TOC_HEADER_FMT = 'BB'
-TOC_ENTRY_FMT = 'BBBix'
-ADDR_FMT = 'BBB' + 'x' * (struct.calcsize('i') - 3)
-CDROMREADTOCHDR = 0x5305
-CDROMREADTOCENTRY = 0x5306
-CDROM_LEADOUT = 0xAA
-CDROM_MSF = 0x02
-CDROM_DATA_TRACK = 0x04
+if sys.platform.startswith('linux'):
+    import linux_cd_parser
+
+try:
+    import discid_parser
+
+    DISCID_AVAILABLE = True
+except ImportError:
+    logger.warn('Cannot import dependency for plugin cd.', exc_info=True)
```

Comment:
I'd prefer if we did not log the exception info here, because the end result looks like an unhandled exception occurred after the warning was displayed (i.e., a traceback in our log is usually a bad news):
```
...
WARNING : Cannot import dependency for plugin cd.
Traceback (most recent call last):
  File "/home/rok/Projects/exaile/plugins/cd/__init__.py", line 52, in <module>
    import discid_parser
  File "/home/rok/Projects/exaile/plugins/cd/discid_parser.py", line 10, in <module>
    import discid
ImportError: No module named discid
...
```

Reply:
Solved: Now it will only give the traceback when executing exaile with debug options enabled.